frontend/amundsen_application/static/js/pages/SearchPage/SearchFilter/InputFilter/index.tsx

```diff
@@ -61,40 +76,112 @@ export class InputFilter extends React.Component<
+    const prevFilterOperation = inputFilter
+      ? inputFilter.filterOperation
+      : undefined;
+    const newFilter = prevFilterOperation
+      ? { value: newValue || undefined, filterOperation: prevFilterOperation }
+      : { value: newValue || undefined };
+
+    const newFilters = {
+      ...filterState,
+      [resourceType]: {
+        ...filterState[resourceType],
+        [categoryId]: newFilter,
+      },
+    };
+    updateFilterState(newFilters);
+  };
+
+  handleFilterOperationChange = (newOperation) => {
+    const {
+      filterState,
+      resourceType,
+      categoryId,
+      updateFilterState,
+    } = this.props;
+
+    this.setState({ filterOperation: newOperation });
+
+    const resourceFilter = filterState[resourceType];
+    const inputFilter = resourceFilter ? resourceFilter[categoryId] : undefined;
+    const prevValue = inputFilter ? inputFilter.value : undefined;
+    const newFilter = prevValue
+      ? { value: prevValue, filterOperation: newOperation }
+      : { filterOperation: newOperation };
 
     const newFilters = {
       ...filterState,
-      [resourceType]: { ...filterState[resourceType], [categoryId]: newValue },
+      [resourceType]: {
+        ...filterState[resourceType],
+        [categoryId]: newFilter,
+      },
     };
     updateFilterState(newFilters);
   };
 
   render = () => {
-    const { value } = this.state;
-    const { categoryId } = this.props;
-    const ariaLabel = categoryId + ' filter input';
+    const { value, filterOperation, showFilterOperationToggle } = this.state;
+    const { categoryId, multiValueSelection } = this.props;
+    const inputAriaLabel = categoryId + 'FilterInput';
+    const toggleGroupId = categoryId + 'Toggle';
     return (
-      <input
-        type="text"
-        className="form-control"
-        name={categoryId}
-        id={categoryId}
-        onChange={this.onInputChange}
-        value={value}
-        aria-label={ariaLabel}
-      />
+      <div>
+        <input
+          type="text"
+          className="form-control"
+          name={categoryId}
+          id={categoryId}
+          onChange={this.onInputChange}
+          value={value}
+          aria-label={inputAriaLabel}
+        />
+        {showFilterOperationToggle && (
+          <div className="filter-operation-toggle">
+            <ToggleButtonGroup
+              type="radio"
+              name={toggleGroupId}
+              id={toggleGroupId}
+              value={filterOperation}
+              onChange={this.handleFilterOperationChange}
+            >
+              <ToggleButton
+                value={FilterOperationType.AND}
+                disabled={!multiValueSelection}
+              >
+                {AND_LABEL}
+              </ToggleButton>
+              <ToggleButton value={FilterOperationType.OR}>
+                {OR_LABEL}
+              </ToggleButton>
+            </ToggleButtonGroup>
+          </div>
```

Comment:
I would probably extract a <FilterOperationToggle /> functional component from this